pkg/envoy/resources.go

```diff
@@ -90,79 +91,91 @@ func (cache *NPHDSCache) OnIPIdentityCacheGC() {
 func (cache *NPHDSCache) OnIPIdentityCacheChange(modType ipcache.CacheModification, cidr net.IPNet,
 	oldHostIP, newHostIP net.IP, oldID *ipcache.Identity, newID ipcache.Identity,
 	encryptKey uint8, k8sMeta *ipcache.K8sMetadata) {
-	// An upsert where an existing pair exists should translate into a
-	// delete (for the old Identity) followed by an upsert (for the new).
-	if oldID != nil && modType == ipcache.Upsert {
-		// Skip update if identity is identical
-		if oldID.ID == newID.ID {
-			return
-		}
-
-		cache.OnIPIdentityCacheChange(ipcache.Delete, cidr, nil, nil, nil, *oldID, encryptKey, k8sMeta)
-	}
-
-	cidrStr := cidr.String()
-
-	scopedLog := log.WithFields(logrus.Fields{
-		logfields.IPAddr:       cidrStr,
-		logfields.Identity:     newID,
-		logfields.Modification: modType,
-	})
-
 	// Look up the current resources for the specified Identity.
 	resourceName := newID.ID.StringID()
 	msg, err := cache.Lookup(NetworkPolicyHostsTypeURL, resourceName)
 	if err != nil {
-		scopedLog.WithError(err).Warning("Can't lookup NPHDS cache")
+		log.WithError(err).Warning("Can't lookup NPHDS cache")
```

Comment:
This change loses the ID being looked up in the logs.

Reply:
reverted.